gdx/src/com/badlogic/gdx/AbstractGraphics.java

```diff
@@ -0,0 +1,33 @@
+package com.badlogic.gdx;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+
+public abstract class AbstractGraphics implements Graphics {
+
+	@Override
+	public void clear(float r, float g, float b, float a){
+		Gdx.gl.glClearColor(r, g, b, a);
+		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
```

Comment:
I'm not sure it's a good idea to have these clear methods, for 3D the standard way is to clear both color and depth, this could be confusing for users and it doesn't add so much value.
Also, i don't think it's the right place for this kind of methods, Graphics is a backend abstraction while clear method wouldn't be more like a utility around GL interface.
I'm in favor to simply remove these 2 clear methods if you don't mind.

Reply:
Ok, I will remove them, the 3D argument is strong.

I've always found that those 2 lines were out of place, in libGDX you don't need to care about the GL interface for anything to do a simple game but you have to copy and paste those 2 ugly strange lines on every screen.  From a non advanced user's perspective, the only thing they do is to clear the screen.

Putting it on a utility class is an option but IMO putting the `Graphics` backend abstraction seems the right place. Clearing the screen every frame seems like part of the `Graphics` functionality any backend would require, in an ideal world (I know it would take more than that) if we had a Non-OpenGL based backend we would also need clear the screen and the `clear` method abstraction would apply.

Reply:
I agree the two lines to clear are cryptic. I think the clear methods are pretty OK on Graphics since clearing the screen is part of the boilerplate for all libgdx apps. It's a valid point that some use cases wouldn't use them, but it doesn't seem terribly confusing to me. FWIW I could go either way, with a lean toward having the convenience methods.

Reply:
Alternatively, you could add a boolean parameter to the method (and then offer an overloaded version of the method with the param set to false), which if true also clears `GL20.GL_DEPTH_BUFFER_BIT`.

Reply:
We all seem to agree it would be good to have the `clear` methods somewhere (with maybe an overloaded version for 3D we can discuss later). 

I've been thinking about it a bit a bit more and I see how `Graphics` is not perfect because it doesn't really deal with rendering stuff, but I'm not sure the alternatives are better.

These are the possible options:
1. Keep it in `Graphics`
2. Utility class. We need some specifics. De we create new class for this? Is there any existing class where this may fit? Are we talking about static utility methods?
3. `GL20`/`GL30` interfaces. If we saw the GLXX interfaces as our Rendering abstraction interface (instead of just a wrapper for OpenGL ES methods), I think this is probably where `clear` would really belong. Alternatively, if we want to keep interfaces that just wrap the OpenGL methods, we can create new Decorator interfaces that have the extra utility methods and are returned by the `Gdx.glXX` methods.

1. and 3. are my preferred options, I probably lean towards 1 for simplicity but I accept 3 is more correct.

@msgx-dev Maybe you can share some more details on how you'd do the utility class option. 

For those who don't like option 1., do you prefer the option 3 alternative?

Reply:
`GLFrameBuffer` could be a reasonable place to put this, we already have a static `unbind` method there and "clearing screen" actually means clearing the current frame buffer, so it'd makes sense.

`ScreenUtils` could be a good place too, class javadoc would need few adaptations but still a good candidate.

`GL20/GL30` would be the worse option IMO. It shouldn't do more than abstracting GLES calls. 

I'm still a bit balanced to put that in `AbstractGraphics`, there is no reasons to override them, these should be static IMO. But if you really want it here, i won't fight against it.

About the depth buffer, a boolean is a good idea, the same way we have one for FrameBuffer constructor: clear(r,g,b,clearDepth) and clear(color,clearDepth). Preferably without overloads, because it's not clear what could be the default. And this way, user knows what he's doing.

I really hope we find a concensus before 2021 :joy: Cheers! :champagne:

Reply:
I think `ScreenUtils` would be a good place if we want it in a utility class, good find! Also, I'm against adding it to `GLFrameBuffer`, as I think even `unbind` shouldn't be static. This would allow overriding it in nestable framebuffer implementations.

Reply:
Ok, thanks for the feedback, I've removed them for now, `ScreenUtils` seems a good fit, we can take the final decision on a separate issue/PR.

Consensus has been achieved before 2021! (in most parts of the globe at least :D)